### PR TITLE
Allow int Subclasses in Query

### PR DESCRIPTION
**Pull request by Exahilosys**
I have found myself having to do explicit type conversions of `int` subclasses that should definitely work.

## What do these changes do?

This new check allows all `int` subclasses through except `bool`, which should obviously be rejected.

## Are there changes in behavior for the user?

Nope. I can't imagine backward compatibility being an issue.

## Checklist

- [X] I think the code is well written
- [X] Unit tests for the changes exist
- [X] Documentation reflects the changes
- [X] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` (e.g. `588.bugfix`)
  * if you don't have an `issue_id` change it to the pr id after creating the PR
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: `Fix issue with non-ascii contents in doctest text files.`

### CODE REVIEW

File: yarl/_url.py [hunk 1]
```diff
@@ -904,7 +904,7 @@ def _query_seq_pairs(cls, quoter, pairs):
     def _query_var(v):
         if isinstance(v, str):
             return v
-        if type(v) is int:  # no subclasses like bool
+        if isinstance(v, int) and not isinstance(v, bool):  # no subclasses like bool
```
Comment (webknjaz) [inline yarl/_url.py:907]: This will probably make it slower so I'll leave it to @asvetlov to decide. But you'd probably have to add tests for this (both negative and positive) and find other places in code that use the same idiom.

Also, what if such `int` subclass implements a weird `__str__()` method that returns something odd? This would create hard-to-debug issues in the calling code, especially if your colleagues don't know YARL internals. Is it really worth it? This could probably be mitigated by an extra check for `v.__str__ is int.__str__` but isn't it overkill?

**Comment by asvetlov** [comment]
Would you describe your use case?

**Comment by Exahilosys** [comment]
@webknjaz I can add tests, please allow for some time as I'm currently on holiday. Also, instead of using the built-in `str` for conversion, `int.__str__` could directly be used (no need to compare `__str__`s then). This does't feel overkill and makes conversion more consistent across the board.

@asvetlov I am making a package with a subclass of `int` meant to represent ids following [twitter's snowflake standard](https://developer.twitter.com/en/docs/basics/twitter-ids) with custom methods for extracting information like worker ids and timestamps. Those ids point to API resources and are sometimes used in the query part of requests.

Since `aiohttp` uses `yarl` under the hood, explicit conversions from that `int` subclass to `str` or `int` are required every time. Situations like this are what this PR is trying to address.

**Comment by Exahilosys** [comment]
I have created both positive and negative tests by restructuring and expanding the current type-checking strategy to a more compact one.  As an extra to the PR, I would also suggest allowing `float` subclasses.

**Comment by serhiy-storchaka** [comment]
The most common `int` subclasses are `bool` and `IntEnum`. `bool` is not allowed in query because of ambiguity of its representation: `1`/`0`, `true`/`false`, `True`/`False`, `yes`/`no`, `on`/`off`, etc. But `IntEnum` also overrides `__str__`, so there is ambiguity of its representation: as `MsgFlag.MSG_CONFIRM` or `2048`? And the representation of standard enums may be changed in Python 3.10, so it will be `socket.MSG_CONFIRM`.

**Comment by asvetlov** [comment]
Fixed by #505

**Comment by asvetlov** [comment]
@Exahilosys thanks for the initial PR!
I've slightly improved it and merged.
